ui: Sinon refactor jest mocks

nathanstilwell opened the pull request:
Refactoring Sinon spies to use Jest mock functions instead. 

Refactored most of the uses of Sinon spies in Cluster UI. There is one
case in `StatementDetails.spec.tsx` where replacing the Sinon spy breaks the
test, so I ended up leaving it for the moment.

There is also a usage of Sinon in `timeScaleDropdown.spec` where
`SinonFakeTimers` are used. Since we are on Jest 27 there is no
replacement for this in Jest. In Jest 28, Jest timers are based on
Sinon fake timers, so when we upgrade we may be able to replace them
then. Or we could keep using Sinon for fake timers.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82908)
<!-- Reviewable:end -->

Commit:
ui: Refactoring Sinon mocks with Jest mock functions

Replacing the uses of Sinon sandbox and spies with Jest's mock functions
(`jest.fn()`).

Refactored most of the uses of Sinon spies in Cluster UI. There is one
case in StatementDetails.spec.tsx where replacing the Sinon spy breaks the
test, so I ended up leaving it for the moment.

There is also a usage of Sinon in timeScaleDropdown.spec where
`SinonFakeTimers` are used. Since we are on Jest 27 there is no
replacement for this in Jest. In Jest 28, Jest timers are based on
Sinon fake timers, so when we upgrade we may be able to replace them
then. Or we could keep using Sinon for fake timers.

Release note: None

Commit:
ui: updating StatementDetails test

Updating test after the removal of the "execution-stats" tab

Release note: None

nathanstilwell commented:
TFTR @matthewtodd!

nathanstilwell commented:
bors r+
